src/configuration.ts

```diff
@@ -65,6 +65,9 @@ export function appDir(): string {
 	try {
 		accessSync(appDir, constants.R_OK | constants.W_OK);
```

Comment:
why does this work on linux? should we move the check somewhere else? seems a little hard since appDir is needed so early in startup, maybe it's needed before we even get to something like `doStartupValidation`

Reply:
I honestly have no idea...it would probably be traced back to the actual c code for the fs calls differing between platforms I suppose. I didn't go digging into fs - it seemed simple enough of a problem to fix.

Reply:
> maybe it's needed before we even get to something like `doStartupValidation`

It would be needed even before validation, gen-config uses it and does zero validation other than this iirc

Reply:
<img width="636" alt="image" src="https://github.com/user-attachments/assets/5cd3d5c2-b5ef-42ec-a5ce-d8568adfb1b3">

Reply:
Then it also affects linux, I wasn't sure if it did, this was the only report of this we have had - probably because of the vast majority of our user-base on v6 being v5 users upgrading.

>have no received reports of this on linux but potentially could apply to it as well.

Reply:
ok fair enough, either way we fixed it